### PR TITLE
Stop loading the entire entity to check user access

### PR DESCRIPTION
**Pull request by almasaeed2010**
Issue #538

## Type(s) of Change(s)

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API-specific change (fix or addition to an API function)
- [ ] Updates documentation (inline or markdown files)

## Description

See issue #538 

This fix preserves backwards compatibility by applying the fix only when an integer is passed to the function instead of an object as expected by the old behavior.

## Testing?

1. add `bio_data/#` links to your main menu (structure -> menus -> main menu -> add link)
1. measure the speed of the page (you can use chrome dev tools. The network tab can show waterfall chart detailing how long a page takes to load)
1. the more links added, the slower the page.

## Additional Notes (if any):

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Using this fix we managed to speed up our site loading by 500%. Cache rebuilding used to take 1.2 minutes while with this fix it only takes 13.35 seconds. Visiting the home page before this fix takes 5.55 seconds after the cache has been repopulated, now it takes 813 milliseconds. This data was collected from testing the fix on our [live site](https://hardwoodgenomics.org).

### CODE REVIEW

File: tripal/includes/TripalEntityUIController.inc [hunk 1]
```diff
@@ -115,7 +115,20 @@ class TripalEntityUIController extends EntityDefaultUIController {
  * @see hook_entity_view_alter()
  */
 function tripal_view_entity($entity, $view_mode = 'full') {
-   $content = '';
+   if(!is_object($entity)) {
+     $id = intval($entity);
+     if($id === 0) {
+       return drupal_not_found();
```
Comment (almasaeed2010) [inline tripal/includes/TripalEntityUIController.inc:121]: Should we return `drupal_not_found()` here or simply return NULL?

**Comment by spficklin** [comment]
I can confirm that this fix works!  I just want to do a look over of the code and then I'll mark my review.

**Comment by chunhuaicheng** [comment]
Thank you @almasaeed2010 for providing a fix so quickly. I can confirm the fix works for me too.

**Review by spficklin — APPROVED** [review]

File: tripal/includes/TripalEntityUIController.inc (hunk 1 above)
Comment (spficklin) [reply tripal/includes/TripalEntityUIController.inc:121]: This seems reasonable to me.

**Comment by spficklin** [comment]
Thanks @almasaeed2010 for the very quick fix on this one!

**Pull request merged**